registry_module.go

```diff
@@ -70,6 +74,66 @@ type RegistryModules interface {
 	UploadTarGzip(ctx context.Context, url string, r io.Reader) error
 }
 
+// TerraformRegistryModule contains data about a module from the Terraform Registry.
+type TerraformRegistryModule struct {
+	ID              string `json:"id"`
+	Owner           string `json:"owner"`
+	Namespace       string `json:"namespace"`
+	Name            string `json:"name"`
+	Version         string `json:"version"`
+	Provider        string `json:"provider"`
+	ProviderLogoURL string `json:"provider_logo_url"`
+	Description     string `json:"description"`
+	Source          string `json:"source"`
+	Tag             string `json:"tag"`
+	PublishedAt     string `json:"published_at"`
+	Downloads       int    `json:"downloads"`
+	Verified        bool   `json:"verified"`
+	Root            `json:"root"`
```

Comment:
Should this be `Root Root ...` ?

Reply:
As written, `Root`'s fields are "promoted" to the top level. That shortens referencing the `Root` of an example `TerraformRegistryModule`, `tfrm` in this case, to `tfrm.Readme` instead of `tfrm.Root.Readme` (the latter also still works, though). 

So I'd say it's a matter of preference, so long as `Root`'s fields don't conflict with those of the parent struct. I'll make it explicitly `Root Root` though.